File: classifier/data/raw/ai/samples/project_euler_java/Problem005.java
/** Project Euler Problem 5: smallest positive number divisible by every integer from 1 to 20. */
public class Problem005 {
    public static void main(String[] args) {
        long multiple = 1;
        for (int n = 2; n <= 20; n++) {
            multiple = lcm(multiple, n);
        }
        System.out.println(multiple);
    }

    private static long lcm(long a, long b) {
        return a / gcd(a, b) * b;
    }

    private static long gcd(long a, long b) {
        while (b != 0) {
            long remainder = a % b;
            a = b;
            b = remainder;
        }
        return a;
    }
}

File: classifier/data/raw/human/samples/project_euler_zig/problem_005.zig
const std = @import("std");

// Project Euler Problem 5: smallest positive number divisible by every integer from 1 to 20.
pub fn main() !void {
    var result: u64 = 1;
    var n: u64 = 2;
    while (n <= 20) : (n += 1) {
        result = result / gcd(result, n) * n;
    }
    try printNumber(result);
}

fn gcd(a: u64, b: u64) u64 {
    var left = a;
    var right = b;
    while (right != 0) {
        const remainder = left % right;
        left = right;
        right = remainder;
    }
    return left;
}

fn printNumber(number: anytype) !void {
    var buffer: [32]u8 = undefined;
    const output = try std.fmt.bufPrint(&buffer, "{d}\n", .{number});
    try std.Io.File.writeStreamingAll(.stdout(), std.Io.Threaded.global_single_threaded.io(), output);
}
